wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -30,6 +37,380 @@ sealed abstract class Wallet extends LockedWallet with UnlockedWalletApi {
+              isInitiator = false,
+              remoteExtPubKey = offer.extPubKey,
+              input = amount,
+              remoteInput = offer.totalCollateral,
+              fundingInputs = fundingUtxos.map(_._1),
+              remoteFundingInputs = offer.fundingInputs,
+              feeRate = offer.feeRate,
+              changeSPK = changeSPK,
+              remoteChangeSPK = offer.changeAddress.scriptPubKey,
+              cetSigs = CETSignatures(PartialSignature.dummyPartialSig,
+                                      PartialSignature.dummyPartialSig,
+                                      PartialSignature.dummyPartialSig),
+              fundingSigs =
+                FundingSignatures(Vector(PartialSignature.dummyPartialSig)),
+              oracleSig = DummySchnorrDigitalSignature,
+              network = network
+            ))
+            .map { _ =>
+              logger.debug("DLC Accept added to database")
+              DLCAccept(
+                amount.satoshis,
+                keyManager.getRootXPub,
+                fundingUtxos,
+                Bech32Address(changeSPK, network),
+                sigs,
+                eventId
+              )
+            }
+        }
+    }
+  }
+
+  override def signDLC(offer: DLCOffer, accept: DLCAccept): Future[DLCSign] = {
+    val clientF = for {
+      account <- getDefaultAccountForType(AddressType.SegWit)
+      txBuilder <- fundRawTransactionInternal(
+        destinations = getDummyDLCFundingOutputs(offer.totalCollateral),
+        feeRate = offer.feeRate,
+        fromAccount = account,
+        keyManagerOpt = Some(keyManager)
+      )
+    } yield {
+      val fundingUtxos =
+        txBuilder.utxoMap.values
+          .flatMap(_.toSingles)
+          .toVector
+      BinaryOutcomeDLCClient.fromOfferAndAccept(offer,
+                                                accept,
+                                                keyManager.rootExtPrivKey,
+                                                fundingUtxos)
+    }
+
+    clientF.flatMap { client =>
+      client.createCETSigs.zip(client.createFundingTransactionSigs()).map {
+        case (cetSigs, fundingSigs) =>
+          DLCSign(cetSigs, fundingSigs, accept.eventId)
+      }
+    }
+  }
+
+  def registerDLCAccept(accept: DLCAccept): Future[DLCDb] = {
+    dlcDAO.findByEventId(accept.eventId).flatMap {
+      case Some(dlcDb) =>
+        logger.debug("DLC found, adding accept data")
+        dlcDAO.update(
+          dlcDb.copy(
+            remoteInput = accept.totalCollateral,
+            remoteExtPubKey = accept.extPubKey,
+            remoteChangeSPK = accept.changeAddress.scriptPubKey,
+            remoteFundingInputs = accept.fundingInputs,
+            cetSigs = accept.cetSigs
+          ))
+      case None =>
+        throw new RuntimeException(
+          s"No DLC found with corresponding eventId ${accept.eventId}")
+    }
+  }
+
+  override def signDLC(accept: DLCAccept): Future[DLCSign] = {
```

Comment:
scaladoc on this and others, make sure to include at what point in the process they are used